tests/integration/client_test.py

```diff
@@ -27,3 +29,25 @@ def test_df(self):
         assert 'Containers' in data
         assert 'Volumes' in data
         assert 'Images' in data
+
+
+class CancellableEventsTest(unittest.TestCase):
+    def test_cancel_events(self):
+        client = docker.from_env()
+        self.addCleanup(client.api.close)
+
+        events = client.events(cancellable=True)
+
+        def cancel_events():
+            time.sleep(1)
+            events.close()
+
+        cancel_thread = threading.Thread(target=cancel_events)
+        cancel_thread.start()
+
+        for _ in events.stream():
+            pass
+
+        events.close()
+
+        # pass, otherwise we wouldn't get here
```

Comment:
maybe we can have a timeout of some kind, so the test doesn't stall forever if we accidentally break something in the future?